src/js/background/background.js

```diff
@@ -274,6 +287,88 @@ async function refreshAccountPages() {
   });
 }
 
+// This function is defined as global in the ESLint config _because_ it is created here:
+// eslint-disable-next-line no-redeclare
+async function makeDomainAddress(address, block_list_emails, description = null) {
+  const apiToken = await browser.storage.local.get("apiToken");
+
+  if (!apiToken.apiToken) {
+    browser.tabs.create({
+      url: RELAY_SITE_ORIGIN,
+    });
+    return;
+  }
+
+  const { relayApiSource } = await browser.storage.local.get("relayApiSource");  
+  const serverStoragePermission = await getServerStoragePref();
+  const relayApiUrlRelayAddress = `${relayApiSource}/domainaddresses/`;
+
+  let apiBody = {
+    "enabled": true,
+    "description": "",
+    "block_list_emails": block_list_emails,
+    "used_on": "",
+    "address": address,
+  };
+
+  // Only send description/generated_for/used_on fields in the request if the user is opt'd into server storage
+  if (description && serverStoragePermission) {
+    apiBody.description = description;
+    apiBody.generated_for = description;
+    apiBody.used_on = description + ",";
+  }
+
+
+  const headers = await createNewHeadersObject({auth: true});
+
+  const newRelayAddressResponse = await fetch(relayApiUrlRelayAddress, {
+    mode: "same-origin",
+    method: "POST",
+    headers: headers,
+    body: JSON.stringify(apiBody),
+  });
+
+  if (newRelayAddressResponse.status === 402) {
+    // FIXME: can this just return newRelayAddressResponse ?
+    return { status: 402 };
+  }
+
+  if (newRelayAddressResponse.status === 409) {
+    return { status: 409 };
+  }
+  
+  if (newRelayAddressResponse.status === 400) {
+    return { status: 400 };
+  }
```

Comment:
CLEAN! Added enhancement in 3938944ab4db86800a065a8164433cb500dcd1c6

Reply:
Hey @maxxcrawford, I saw "CLEAN" and wanted to check it out, but just a heads-up in case it was unintentional: https://github.com/mozilla/fx-private-relay-add-on/commit/3938944ab4db86800a065a8164433cb500dcd1c6 changes the behaviour. Whereas before you returned a `{ status: number }`, you're now just returning a `number`. Not sure if the calling code expects that. And there's also a `console.log` in there.

The comment describing the error codes is excellent btw :)

Reply:
@Vinnl EAGLE EYE! Revised in 76c1793ee8d3fde02b3f46152064d95600c65523

Reply:
Teamwork!